### ISSUE TITLE
Decryption of accounts in Neo 2 NEP6 wallets fails using @cityofzion/neon-core version ^5.0.0-next.4

### ISSUE THREAD

**Issue by djnicholson** [CityOfZion/neon-js#605]
I'm currently prototyping a Neo 3 compatible version of the [Visual DevTracker](https://github.com/neo-project/neo-visual-tracker).

I'm using the following packages:
```
    "@cityofzion/neon-core": "^5.0.0-next.4",
    "@cityofzion/neon-js": "^5.0.0-next.4",
```

I'm seeing that the `decrypt` method on the `wallet.Wallet` class always returns the error `Wrong password or scrypt parameters!` when operating on a Neo 2 wallet (it works fine with Neo 3 wallets).  I am using the correct password.  The scrypt parameters seem to be the same regardless of whether a wallet is generated using `@cityofzion/neon-core` version `4.8.1` or version `5.0.0-next.4` so perhaps the error text is misleading.

**Comment by snowypowers** [comment]
The encryption/decryption of the key relies on the address generation mechanism as a checksum. In NEO3, the address header was changed. For this, I have included the method `decryptNeo2` under the `wallet` package. Please try that if you are looking to transit wallets.

We can probably update the error message if we are decrypting through the `wallet.Wallet` as we should have access to the address and will be able to discern if the key is NEO2.

tldr: Encrypted keys will be different between versions. Use compatibility method `decryptNeo2` to decrypt old wallets.

**Comment by djnicholson** [comment]
Thanks for the explanation, @snowypowers!

It would be really useful for me if `packages/neon-core/src/wallet/Account.ts` could contain some logic to know whether it represents a V2 or V3 account (it could look at the address prefix to determine this) and then in the Account.decrypt method call either `decrypt` or `decryptNeo2` as appropriate.

My ultimate goal is to have one package that can operate on both Neo 2 and Neo 3 wallets while only depending on one version of  `@cityofzion/neon-core`.  (Tell me if that's not the direction you are going with the package.)

If you agree with the approach I'd be happy to create a PR.

**Comment by snowypowers** [comment]
Directly calling `decryptNeo2` from the Account class has issues as it does not migrate the current Account instance to NEO3. The addresses are very different in Neo3 so having you decrypt the Neo2 Account instance might just cause unnecessary confusion (the `address` data is prefilled so it is not always generated from the private key when available, the verification script will also look different but it is not adjusted during the decryption process).

There is a `upgrade` method in wallet/upgrade.ts that can be used to create a Neo3 Account instance from a Neo2 Account instance.

There is no intention to be able to sign both neo2 and neo3 transactions with the same version. My current intention is to only provide an upgrade solution for the keys.
